src/libstd/sys/unix/backtrace.rs

```diff
@@ -133,7 +133,17 @@ pub fn write(w: &mut Write) -> io::Result<()> {
     result::fold(iter, (), |_, _| ())
 }
 
-#[cfg(not(all(target_os = "ios", target_arch = "arm")))]
+#[cfg(target_os = "nacl")]
+#[inline(never)]
+pub fn write(_w: &mut Write) -> io::Result<()> {
+    use io::ErrorKind;
+    Err(io::Error::new(ErrorKind::PermissionDenied,
+                       "can't read the stack instruction pointer \
+                        to find the backtrace"))
+}
+
+#[cfg(not(any(all(target_os = "ios", target_arch = "arm"),
+              target_os = "nacl")))]
```

Comment:
I think this module may want a bit more of a principled approach at this point, the organization in here is getting pretty hairy from place to place. Could you organize this with something like:

```
backtrace/libbacktrace.rs
backtrace/noop.rs
backtrace/os.rs
backtrace/mod.rs
```